### ISSUE TITLE
Support F#

### ISSUE THREAD

**Issue by pbiggar** [darklang/tablecloth#196]
I'm working on adding support for F#. Although it wouldn't be identical, it would have the same function names as in Rescript and OCaml. The advantages of this are:

- consistency with libraries using Tablecloth elsewhere
- F# has many of the tablecloth functions implemented as methods, which make them hard to pipe
- some of the functions are implemented in one of FSharpX or FSharpPlus libraries, but not all of tablecloth's functions are implemented in either one

I'm currently doing this in a branch at https://github.com/darklang/dark/commits/paul/more-apiserver3, and I plan to move it here soon.

**Comment by pbiggar** [comment]
The completed F# version is here: https://github.com/darklang/dark/tree/main/fsharp-backend/src/Prelude

**Comment by pm5** [comment]
Any plan to backport from dark?

**Comment by pbiggar** [comment]
Yes, but not a priority at this moment. Intending to release the main branch as 0.0.8, and then move to the `next` branch, to which we'll add F#. Feel free to make a PR against the `next` branch incorporating the code from the dark repo.

**Comment by pbiggar** [comment]
Next steps here:

- copy code from Dark repo
- port tests

My thinking for the tests is that there should be a json file with function names, inputs and outputs, so that they can be the same for each language. (we'd probably need some exceptions at some point, and this would help us learn what exceptions are needed)

**Comment by pbiggar** [comment]
Partially done via #296. Next steps:

- [ ] add build/etc to makefile
- [ ] add build to CI
- [ ] add tests
